pages/admin/users.vue

```diff
@@ -1,13 +1,279 @@
 <script setup lang="ts">
-  const { data } = await useFetch('/api/user?scope=all')
+  import { usertypes as usertypeNames } from '~/config/app.config'
+  import type { User } from '~/models/User'
+
+  const { data: users, refresh } = await useFetch<User[]>('/api/user?scope=all')
+  const { data: companies } = await useFetch('/api/company')
+
+  const { text, copy, copied } = useClipboard()
+
+  const usertypes = ref([0, 1, 2])
+  const filterOrder = ref('descending')
+  const filterType = ref('updated')
+
+  const selected = ref([])
+
+  const pageSize = 10
+  const currentPage = ref(1)
+
+  const loading = ref()
+  const dialog = ref(false)
+
+  const filteredUsers = computed(() => {
+    // eslint-disable-next-line
+    const selectedUsertypes = usertypeNames.filter((usertype, index) =>
+      usertypes.value.includes(index)
+    )
+
+    const filteredUsers = users.value?.filter((user) =>
+      selectedUsertypes.includes(user.userType ?? 'basic')
+    )
+
+    selected.value = []
+
+    return (
+      filteredUsers?.sort((a, b) => {
+        /* @ts-ignore */
+        const order = a[filterType.value] < b[filterType.value] ? -1 : 1
+        return filterOrder.value === 'ascending' ? order : order * -1
+      }) ?? []
+    )
+  })
+
+  // from stackoverflow
+  const pages = computed(() =>
+    filteredUsers.value.reduce((all, one, i) => {
+      const ch = Math.floor(i / pageSize)
+      all[ch] = [].concat(all[ch] || [], one)
+      return all
+    }, [])
+  )
+
+  const deleteUsers = async () => {
+    loading.value = true
+
+    const queries = []
+
+    const selectedUsers = filteredUsers.value.map((user, index) =>
+      selected.value[index] ? user.uid : undefined
+    )
+
+    for (const uid of selectedUsers)
+      !uid ||
+        queries.push(
+          $fetch('/api/user', {
+            method: 'DELETE',
+            body: {
+              uid,
+            },
+          })
+        )
```

Comment:
Interesting. Let's hope there is not too many users selected.
Jokes aside, I assume this will be improved when the user delete request enables deletion of multiple users at once, so we will allow it.

Reply:
Haha ye, I'll update the API-endpoint to allow deleting multiple users at once.